.chloggen/remove_top_level_block.yaml

```diff
@@ -0,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: confmap
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Remove top level lock when considering struct as Unmarshalers
```

Comment:
This says a top level lock is removed, but I don't see that in the change. Can you point me to where this lock is removed?

Reply:
here: https://github.com/open-telemetry/opentelemetry-collector/pull/9862/files#diff-3217721f23d028519f5993ab4e56eaff85988734dab4068a7eb411b838cd2786R331

Reply:
@atoulme I don't see a lock in that snippet.

Or are you saying by adding `skipTopLevelUnmarshaler` to the condition you now allow invokers to choose to unmarshal to level struct?

Reply:
I have renamed the PR from using the term "lock" to "condition". Hopefully that clarifies a bit.

Reply:
Thanks @atoulme!